Rock/Model/CodeGenerated/ReportService.cs

```diff
@@ -53,6 +54,43 @@ public bool CanDelete( Report item, out string errorMessage )
             errorMessage = string.Empty;
             return true;
         }
+
+        /// <summary>
+        /// Create a new non-persisted Report using an existing Report as a template. 
+        /// </summary>
+        /// <param name="reportId">The identifier of a Report to use as a template for the new Report.</param>
+        /// <returns></returns>
+        public Report GetNewFromTemplate( int reportId )
```

Comment:
@MrUpsideDown , I didn't catch this when I pulled this, but you want to avoid modifying any of the files in the codegenerated files. We periodically regenerate these files and any changes to them would get lost. They are all partial classes so anytime you need to add methods to them (i.e. service classes ), you can create a new partial class ( see this commit that fixes this request for an example: https://github.com/SparkDevNetwork/Rock/commit/27ae302b1a8841c626bcfd59d98ad4c3ce0debe1 )